Distribution/Server/Pages/Recent.hs

```diff
@@ -176,17 +176,19 @@ revisionItem :: Users -> URI -> PkgInfo -> [RSS.ItemElem]
 revisionItem users hostURI pkgInfo =
   [ RSS.Title title
   , RSS.Link uri
-  , RSS.Guid True (uriToString id uri "")
+  , RSS.Guid True (uriToString id guid "")
   , RSS.PubDate time
   , RSS.Description desc
   ]
   where
     uri   = hostURI { uriPath = packageURL pkgId  ++ "/revisions"}
+    guid  = hostURI { uriPath = packageURL pkgId  ++ "/revision/" ++ show revision }
     title = display (packageName pkgId) ++ " " ++ display (packageVersion pkgId)
-    body  = "Revision #" ++ show (pkgNumRevisions pkgInfo - 1)
+    body  = "Revision #" ++ show revision
     desc  = "<i>Revised by " ++ display user ++ ", " ++ showTime time ++ ".</i>"
          ++ if null body then "" else "<p>" ++ body
     user = Users.userIdToName users userId
+    revision = pkgNumRevisions pkgInfo - 1
```

Comment:
I haven't looked at the code, but this one looks odd; is this always non-negative?